app/src/main/java/com/example/nvdovin/weatherapp/ForecastActivity.java

```diff
@@ -30,27 +37,35 @@ protected void onCreate(Bundle savedInstanceState) {
         RecyclerView recyclerView = (RecyclerView) findViewById(R.id.recycler_view);
         progressBar = (ProgressBar) findViewById(R.id.forecast_progress_bar);
 
-        Constants constants = new Constants();
-        int CELSIUS_SCALE = constants.getCelsiusScale();
-        List<City> cityList = new ArrayList<City>();
 
+        List<City> cityList = new ArrayList<City>();
         TypedValue outValue = new TypedValue();
-
         getResources().getValue(R.dimen.separator_height, outValue, true);
 
+        sharedPrefs = new SharedPrefs(this);
+
+        swipeRefreshLayout = (SwipeRefreshLayout) findViewById(R.id.forecast_swipe_refresh);
+        swipeRefreshLayout.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
+            @Override
+            public void onRefresh() {
+                refreshProcedure();
+            }
+        });
+
         float separatorHeight = outValue.getFloat();
         SeparatorDecoration separatorDecoration = new SeparatorDecoration(this, Color.GRAY, separatorHeight);
+
         RetrofitFactory retrofitFactory = new RetrofitFactory();
-        GreenDaoFactory greenDaoFactory = new GreenDaoFactory(this);
+        greenDaoFactory = new GreenDaoFactory(this);
 
         ForecastPresenter forecastPresenter = new ForecastPresenter(retrofitFactory, greenDaoFactory, this);
```

Comment:
what is the point of creating a local variable and then assign it to global variable???